zaza/charm_tests/vault/utils.py

```diff
@@ -203,9 +203,8 @@ def auth_all(clients, token):
 
 
 def run_charm_authorize(token):
-    unit = zaza.model.get_first_unit_name(utils.get_juju_model(), 'vault')
-    return zaza.model.run_action(
+    return zaza.model.run_action_on_leader(
         utils.get_juju_model(),
-        unit,
+        'vault',
```

Comment:
Would it be useful to pass charm_name in from zaza that has that information from tests.yaml? Not requesting that this change need to contain that, but we might want to have that. I have a example of how we could do that in my generic actions WIP.